app/components/transUnit/TransUnitService.js

```diff
@@ -22,10 +21,16 @@
 
     // TODO can move or delegate to PhraseUtil
     transUnitService.isTranslationModified = function(phrase) {
-      // on Firefox with input method turned on,
-      // when hitting tab it seems to turn undefined value into ''
-      return nullToEmpty(phrase.newTranslation) !== nullToEmpty(
-          phrase.translation);
+      if (phrase.plural) {
+        // on Firefox with input method turned on,
+        // when hitting tab it seems to turn undefined value into ''
+        return nullToEmpty(phrase.translations.toString()) !==
+          nullToEmpty(phrase.newTranslations.toString());
```

Comment:
Converting the arrays to strings for comparison does not seem like good style to me (it is inefficient and may have edge cases where it fails). It is a bit shorter than other options, but lodash makes it pretty easy to compare each value:

```
var allSame = _.every(phrase.translations, function(translation, index) {
  return translation === phrase.newTranslations[index];
});
return !allSame;
```